src/components/Blogs.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+import axiosInstance from "../api/axios";
+
+const Blogs = () => {
+  const [posts, setPosts] = useState([]);
+
+  const getBlog = async () => {
+    try {
+      const results = await axiosInstance.get();
+      setPosts(results.data.results);
+      // console.log(results.data.getPosts)
+      // console.log(results.data.results)
+    } catch {
+      console.log("Error getting blogs");
+    }
+  };
+
+  useEffect(() => {
+    getBlog();
+  }, []);
```

Comment:
Should `getBlog` not be defined inside the `useEffect`?

I am of the opinion that `React` must have screamed at you for not adding `getBlog` to the `useEffect`'s dependency array

Reply:
calling it inside the useEffect hook is also perfect, I also learnt declaring the variables or functions outside the useEffect helps because the variable won't get recreated every time the App component is re-rendered as useEffect does not need to keep track of it in its dependencies array.

Reply:
But in your current case, whenever this file rerenders, `getBlog` is recreated, that's why you would need to add it to the dependency array.

Creating the function inside the useEffect(with an empty dependency array), would cause React to only define that function once.

Reply:
ok will be adjusted